solidity/contracts/liquidity-protection/LiquidityProtectionSettings.sol

```diff
@@ -0,0 +1,443 @@
+     *
+     * @param _prevMinNetworkCompensation  previous minimum network token compensation
+     * @param _newMinNetworkCompensation   new minimum network token compensation
+     */
+    event MinNetworkCompensationUpdated(uint256 _prevMinNetworkCompensation, uint256 _newMinNetworkCompensation);
+
+    /**
+     * @dev triggered when the network token lock duration is updated
+     *
+     * @param _prevLockDuration  previous network token lock duration, in seconds
+     * @param _newLockDuration   new network token lock duration, in seconds
+     */
+    event LockDurationUpdated(uint256 _prevLockDuration, uint256 _newLockDuration);
+
+    /**
+     * @dev triggered when the maximum deviation of the average rate from the spot rate is updated
+     *
+     * @param _prevAverageRateMaxDeviation previous maximum deviation of the average rate from the spot rate
+     * @param _newAverageRateMaxDeviation  new maximum deviation of the average rate from the spot rate
+     */
+    event AverageRateMaxDeviationUpdated(uint32 _prevAverageRateMaxDeviation, uint32 _newAverageRateMaxDeviation);
+
+    /**
+     * @dev initializes a new LiquidityProtectionSettings contract
+     *
+     * @param _registry contract registry
+     * @param _networkToken the network token
+     */
+    constructor(IERC20Token _networkToken, IContractRegistry _registry)
+        public
+        ContractRegistryClient(_registry)
+        validAddress(address(_networkToken))
+        notThis(address(_networkToken))
+    {
+        // set up administrative roles.
+        _setRoleAdmin(ROLE_OWNER, ROLE_OWNER);
+        _setRoleAdmin(ROLE_WHITELIST_ADMIN, ROLE_OWNER);
+
+        // allow the deployer to initially govern the contract.
+        _setupRole(ROLE_OWNER, msg.sender);
+
+        networkToken = _networkToken;
+    }
+
+    modifier onlyOwner() {
+        _onlyOwner();
+        _;
+    }
+
+    // error message binary size optimization
+    function _onlyOwner() internal view {
+        require(hasRole(ROLE_OWNER, msg.sender), "ERR_ACCESS_DENIED");
+    }
+
+    modifier onlyWhitelistAdmin() {
+        _onlyWhitelistAdmin();
+        _;
+    }
+
+    // error message binary size optimization
+    function _onlyWhitelistAdmin() internal view {
+        require(hasRole(ROLE_WHITELIST_ADMIN, msg.sender), "ERR_ACCESS_DENIED");
+    }
+
+    // ensures that the portion is valid
+    modifier validPortion(uint32 _portion) {
+        _validPortion(_portion);
+        _;
+    }
+
+    // error message binary size optimization
+    function _validPortion(uint32 _portion) internal pure {
+        require(_portion > 0 && _portion <= PPM_RESOLUTION, "ERR_INVALID_PORTION");
+    }
+
+    /**
+     * @dev adds a pool to the whitelist
+     * can only be called by the contract owner
+     *
+     * @param _poolAnchor pool anchor
+     */
+    function addPoolToWhitelist(IConverterAnchor _poolAnchor)
+        external
+        override
+        onlyWhitelistAdmin
```

Comment:
that's a logical change - was previously whitelist admin or owner - intentional?

Reply:
You're right. Yes, I wanted to be less verbose, since the owner can always set himself as a whitelist admin

Reply:
ok, cool